src/libvirt-xml-create.js

```diff
@@ -249,22 +249,29 @@ export function getSnapshotXML(name, description, disks, memoryPath, isExternal,
             // Disk can have attribute "snapshot" set to "no", which means no snapshot should be created of the said disk
             // This cannot be configured through cockpit, but we should uphold it nevertheless
             // see "snapshot" attribute of <disk> element at https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms
-            if (disk.snapshot !== "no") {
-                const diskElem = doc.createElement('disk');
-                diskElem.setAttribute('name', disk.target);
-                diskElem.setAttribute('snapshot', 'external');
-
-                if (disk.type === "volume") {
-                    const poolPath = getStoragePoolPath(storagePools, disk.source.pool, connectionName);
-                    if (poolPath) {
-                        const sourceElem = doc.createElement('source');
-                        sourceElem.setAttribute('file', `${poolPath}/${disk.source.volume}.snap`);
-                        diskElem.appendChild(sourceElem);
-                    }
+            if (disk.snapshot == "no")
+                return;
+
+            // Disks of type "file" but without a source (such as
```

Comment:
This explanation is wrong. For external snapshots, all disks that should be part of the snapshots must be explicitly listed, even on the command line. There is nothing "automatic". We just need to skip it here.